FBTweak/FBTweak.h

```diff
@@ -37,132 +37,167 @@ typedef id FBTweakValue;
  */
-@property (nonatomic, copy, readwrite) NSString *name;
+@protocol FBActionTweak <FBTweak>
 
 /**
-  @abstract If this tweak is an action, with a block value.
-  @param If YES, {@ref currentValue} should not be set and
-    {@ref defaultValue} is a block rather than a value object.
+ @abstract The underlying block for this tweak.
  */
-@property (nonatomic, readonly, assign, getter = isAction) BOOL action;
+@property (readonly ,nonatomic) dispatch_block_t currentValue;
+
+@end
 
 /**
-  @abstract The default value of the tweak.
-  @discussion Use this when the current value is unset.
-    For actions, set this property to a block instead.
+ @abstract Represent an editable \c FBTweak, whose \c currentValue can be set.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue defaultValue;
+@protocol FBEditableTweak <FBTweak>
 
 /**
-  @abstract The current value of the tweak. Can be nil.
-  @discussion Changes will be propagated to disk. Enforces within
-    possible values when changed. Must not be set on actions.
+ @abstract The current value of the tweak.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue currentValue;
+@property (readwrite, nonatomic, nullable) FBTweakValue currentValue;
 
 /**
-  @abstract The possible values of the tweak.
-  @discussion Optional. If nil, any value is allowed. If an
-    FBTweakNumericRange, represents a range of numeric values.
-    If an array or dictionary, contains all of the allowed values.
-    Should not be set on tweaks representing actions.
+ @abstract The default value of the tweak. It is up to the user whether to use this value when the
+ \c currentValue is \c nil.
  */
-@property (nonatomic, strong, readwrite) id possibleValues;
+@property (readonly ,nonatomic) FBTweakValue defaultValue;
 
 /**
-  @abstract The minimum value of the tweak.
-  @discussion Optional. If nil, there is no minimum. Numeric only.
-    Should not be set on tweaks representing actions.
+ @abstract Reset the tweak to its initial state.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue minimumValue;
+- (void)reset;
 
 /**
-  @abstract The maximum value of the tweak.
-  @discussion Optional. If nil, there is no maximum. Numeric only.
-    Should not be set on tweaks representing actions.
+ @abstract The possible values of the tweak.
+ @discussion If nil, any value is allowed. If an FBTweakNumericRange, represents a range of numeric
+ values. If \c NSSArray or \c NSDictionary, contains all of the allowed values.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue maximumValue;
+@property (readwrite, nonatomic, nullable) id possibleValues;
 
 /**
-  @abstract The step value of the tweak.
-  @discussion Optional. If nil, the step value is calculated from
-    the miniumum and maxium values. Only used for numeric tweaks.
+ @abstract The minimum value of the tweak.
+ @discussion If \c nil, there is no minimum. Applicable only when \c currentValue is expected to
```

Comment:
Why the `expected to`?
(Same everywhere)

Reply:
Done